core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1733,7 +1733,12 @@ public function generateRecordLabel(array $row, string $table = null, bool $prot
 						$args_k[] = $GLOBALS['TL_DCA'][$table]['fields'][$v]['reference'][$option] ?? $option;
 					}
 
-					$args[$k] = implode(', ', $args_k);
+					$implode = static function ($v) use (&$implode)
+					{
+						return implode(', ', array_map(static fn ($vv) => \is_array($vv) ? $implode($vv) : $vv, $v));
+					};
+
+					$args[$k] = $implode($args_k);
```

Comment:
How about using a recursive iterator here?

```suggestion
					$args[$k] = implode(', ', iterator_to_array(new \RecursiveIteratorIterator(new \RecursiveArrayIterator($args_k)), false));
```

Reply:
The recursive iterator is slightly slower (see https://3v4l.org/X3j8B and https://3v4l.org/PWOI8) and the result is different:

```php
$a = [
    'foo',
    'bar' => [
        'foo1',
        'bar1' => [
            'foo2',
            'bar2' => [],
        ],
        'baz1',
    ],
    'baz',
];
```

- `$implode` -> `foo, foo1, foo2, , baz1, baz`
-  iterator -> `foo, foo1, foo2, baz1, baz`

Which value is the "correct" one?

Reply:
I don’t think speed is relevant here.
The output of the iterator version is “correct” IMO.